### PR TITLE
Bring datepicker props inline with their description

### LINKED ISSUE #8650: DatePicker(disableAutoFocus=true) doesn't auto move focus to calendar when triggering out calendar manually
## Codepen demonstrating the issue
https://codepen.io/yudma/pen/Pgbxpy

## Actual behavior:
when setting `disableAutoFocus=true`, opening the calendar manually by pressing enter, user need to press tab again to move focus into the calendar pop up.
## Expected behavior:
image
When pressing enter to manually open calendar, focus automatically moves into calendar.
P.S. According to the screenshot above, the `DisableAutoFocus` prop indicates disable **auto-popup** of calendar, but actually, it at the same time disables **auto move focus to calendar** when manually pop up calendar. Then user needs to press tab to focus in.

## Documentation describing expected behavior
Don't know where to find the exact document, but OUFR Dialog, Callout also follow the pattern that focus auto-moves into the popped up UI.
We can see this pattern in their demo pages:
https://developer.microsoft.com/en-us/fabric#/components/dialog
https://developer.microsoft.com/en-us/fabric#/components/callout

### PR DESCRIPTION
**Pull request by joschect**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #8650
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Previously the disableAutoFocus prop would cause the datepicker popup to not get focused, even though it's description just states that the menu should not open when the input button is focused. This pr brings the behavior inline with that comment and adds another prop to provide support for the existing behavior where the callout is not focused if it's set to true
#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8662)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts [hunk 1]
```diff
@@ -115,6 +115,12 @@ export interface IDatePickerProps extends IBaseProps<IDatePicker>, React.HTMLAtt
    */
   disableAutoFocus?: boolean;
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts]: does the old prop need to be deprecated?

Comment (joschect) [reply packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts]: No, it now does what it says it's going to do. It needs to be kept around!

**Comment by ecraig12345** [comment]
Is there any way we can handle this scenario without adding a new prop for a special case? Seems like a bit of a code smell.

**Comment by joschect** [comment]
@ecraig12345 We could maybe do something like this by making an composited callout that automatically sets focus and takes in that prop so that we aren't explicitly adding it to the datepicker, in that case the prop would be passed to the callout instead. Something like that could work. I'll see if I can make it do that. We could try just removing that prop all together, it might not be necessary behavior.

**Comment by ecraig12345** [comment]
> We could try just removing that prop all together, it might not be necessary behavior.

Since it's a public API you'd have to deprecate not remove. And before doing that, I'd recommend looking at the history to see why it was added.

**Comment by joschect** [comment]
@ecraig12345 I meant remove the prop that I added, technically we don't need it.

File: packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.base.tsx [hunk 2]
```diff
@@ -296,7 +296,14 @@ export class DatePickerBase extends BaseComponent<IDatePickerProps, IDatePickerS
   };
 
   private _onCalloutPositioned = (): void => {
-    if (this._calendar.current && !this.props.disableAutoFocus) {
+    let shouldFocus = true;
+    // If the user has sepecified that the callout shouldn't use initial focus, then respect
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.base.tsx]: specified

**Comment by joschect** [comment]
@ecraig12345 This should be ready to go in, could you give it another look?

**Comment by dzearing** [comment]
This looks good and issues above have been addressed.

**Pull request merged**